### DISCUSSION TITLE
Can you use WDIR&WIND observations to calculate VCNT?

### DISCUSSION THREAD

**Discussion by heu3becteh** [dtcenter/METplus#2240] [File I/O] [answered]
Hello!

We have UGRD and VGRD for forecast data and WDIR and WIND for observations.
When calculating wind speed statistics, MET does successfully convert from U/V components to wind speed.
```
{
  name       = "WIND";
  level      = [ "Z10" ];
  cat_thresh = [ >2, >5, >12, >12.5, >15, >20, >30 ];
  mpr_column = [ "ABS(OBS-FCST)" ];
  mpr_thresh = [ <=50 ];
}
```
But it does not read WDIR data.

When we try to calculate Vector Statistics, we have an empty *_vcnt.txt files (with header only, no data).

Is it possible to use observation data as is or do I have to recalculate WDIR&WIND to U&V components externally/manually (which does work) before using MET?

As I understand, when not using `obs = fcst;` each obs variable entry is verified with corresponding fcst variable, so configuration like that
```
fcst = {
  message_type = [ "ADPSFC" ];

  field = [
    {
      name       = "UGRD";
      level      = [ "Z10" ];
      is_u_wind  = TRUE;
    },
    {
      name       = "VGRD";
      level      = [ "Z10" ];
      is_v_wind  = TRUE;
    }
  ];
};

obs = {
  message_type = [ "ADPSFC" ];

  field = [
    {
      name       = "WDIR";
      level      = [ "Z10" ];
      is_wind_direction = TRUE;
    },
    {
      name       = "WIND";
      level      = [ "Z10" ];
      cat_thresh = [ >2, >5, >12, >12.5, >15, >20, >30 ];
      mpr_column = [ "ABS(OBS-FCST)" ];
      mpr_thresh = [ <=50 ];
      is_wind_speed = TRUE;
    }
  ];
};
```
does not work, as it will match UGRD with WDIR and VGRD with WIND. Is it possible to use different formats of wind input data?

I do get that it is not possible to verify WDIR itself from [this topic](https://github.com/dtcenter/METplus/discussions/1738), but it would be useful to be able to use U&V fcst together with WDIR&WIND obs data as input, as it is the most common format, I assume.

Best regards.

**Comment by JohnHalleyGotway** [comment, marked as answer]
@heu3becteh, unfortunately, the answer is no. Point-Stat isn't set up to pair U/V forecasts with wind speed/direction observations to compute the wind vector line types (VL1L2, VAL1L2, and/or VCNT). I can see how the logic could potentially be revised in the future to support that... but it doesn't do it now.

With the current capabilities, you could use python embedding for the point observations. The python script could read the wind speed/direction, convert to U/V, and pass the result to Point-Stat. But I realize that does complicate matters.

Is this a challenge you've been experiencing for a while? Wondering if I should write up an issue in GitHub to request this new feature be added to a future version?

**Reply by heu3becteh** [reply]
@JohnHalleyGotway, thank you for the reply!

It is not complicated to convert wind direction and speed to U/V when writing the observation data, I will do that. I guess there is no reason to store WDIR&WIND in the observation data at all, but maybe I will leave these too.

I was hoping I didn't use `is_wind_direction` and `is_wind_speed` flags the right way and they will help to use the wind data as is, it could be a useful feature, but it is not much of a problem without it.

**Reply by JohnHalleyGotway** [reply]
Thanks for selecting an answer on this discussion. FYI, after talking to scientists on the METplus team, I did write up this feature request for MET as issue dtcenter/MET#2607. I'll go ahead and lock this discussion to prevent future posts, but please feel free to comment on that MET issue as needed.
